src/QMCHamiltonians/NonLocalECPComponent.cpp

```diff
@@ -24,6 +24,16 @@ namespace qmcplusplus
 {
 NonLocalECPComponent::NonLocalECPComponent() : lmax(0), nchannel(0), nknot(0), Rmax(-1), VP(nullptr) {}
 
+// unfortunately we continue the sloppy use of the default copy constructor followed by reassigning pointers.
+// This prevents use of smart pointers and concievably sets us up for trouble with double frees and the destructor.
+NonLocalECPComponent::NonLocalECPComponent(const NonLocalECPComponent& nl_ecpc, const ParticleSet& pset) : NonLocalECPComponent(nl_ecpc)
```

Comment:
NonLocalECPComponent doesn't need inheritance and thus a specialized copy constructor is better. Yes a bit unfortunate that we would like to to minimal change.

Reply:
The alternative is to have the incorrect refPS for the NonLocalECPComponent contained VP set. Seems like that would result in the wrong distance tables being used and infact a possible race on the differerent tables.

Reply:
The real solution to this is that no object constructed via cloning of the golden walker elements should retain any reference to another golden element and arguments needed for an evaluation from other elements need to be explicitly in the API.